### ISSUE TITLE
vmselect: /api/v1/export does not return all matching metrics when start/end params are used

### ISSUE THREAD

**Issue by immerrr** [VictoriaMetrics/VictoriaMetrics#2079]
**Describe the bug**

I have a rather small single-server deployment of VM v1.71.0 (1.6B datapoints, ~30G data). I have noticed that some data was missing in the outputs from time to time. I narrowed it down to a reproduceable `/export` endpoint which does not return some of the metrics that are present in storage if `start`/`end` query params are used.

**To Reproduce**

I have restored production data from a backup into a local instance (built from sources), and I am running locally an export query like this
```
/api/v1/export?match[]=my_metric{metristor_ns="default",tag1="val1",tag2="val2"}&start=1633046400.0&end=1640908800.0
```

and it returns 5 metrics. If I repeat the request, but omit the start/end values, 

```
/api/v1/export?match[]=my_metric{metristor_ns="default",tag1="val1",tag2="val2"}
```

it returns 10 metrics, and those 5 metrics not returned previously **all** have some data for the 1633046400.0-1640908800.0 interval.

NOTE: I have tried exporting the metrics in question and *was unable to reproduce the issue on a clean VictoriaMetrics instance after import*. It appears that the issue is related to the internal state (of the backup) somehow, but unfortunately, I am not able to share the data itself as of now. I am happy to run further tests locally.

**Expected behavior**

I expect `/export` endpoint with filtering by `start/end` to return timeseries that match the `match[]` param and have data in `start/end` interval.

**Logs**

The logs only show information regarding loading of parts, and nothing related to the query per se. Since I was able to reproduce locally with VictoriaMetrics built from sources, I was able to narrow the difference in code paths down to [a section in `indexDB.searchTSIDs` function](https://github.com/immerrr/victoriametrics/blob/4701c108ff98a3c935c0be4791379fc674162f01/lib/storage/index_db.go#L1636-L1638):

```
	fmt.Printf("indexDB.searchTSIDS tfss=%+v\n", tfss)
	fmt.Printf("indexDB.searchTSIDS tr=%+v\n", tr)
	fmt.Printf("db.s.minTimestampForCompositeIndex=%+v\n", db.s.minTimestampForCompositeIndex)
	if tr.MinTimestamp >= db.s.minTimestampForCompositeIndex {
		fmt.Printf("convertToCompositeTagFilterss...\n")
		tfss = convertToCompositeTagFilterss(tfss)
	}
```

It prints the following for cases with start/end filtering when not all metrics are returned:
```
indexDB.searchTSIDS tfss=[{__name__="my_metric", metristor_ns="default", tag1="val1", tag2="val2"}]
indexDB.searchTSIDS tr={MinTimestamp:1618704000000 MaxTimestamp:1640908800000}
db.s.minTimestampForCompositeIndex=1618704000000
convertToCompositeTagFilterss...
indexSearch.searchTSIDs returned localTSIDs=5
```

And the following for the case without start/end filtering
```
indexDB.searchTSIDS tfss=[{__name__="my_metric", metristor_ns="default", tag1="val1", tag2="val2"}]
indexDB.searchTSIDS tr={MinTimestamp:0 MaxTimestamp:1642522266000}
db.s.minTimestampForCompositeIndex=1618704000000
indexSearch.searchTSIDs returned localTSIDs=10
```

To verify that the issue is related to composite tags, I have run a request with `start=1618703000.0` and the same `end=1640908800.0`, and it returned all metrics as expected:

```
indexDB.searchTSIDS tfss=[{__name__="my_metric", metristor_ns="default", tag1="val1", tag2="val2"}]
indexDB.searchTSIDS tr={MinTimestamp:1618703000000 MaxTimestamp:1640908800000}
db.s.minTimestampForCompositeIndex=1618704000000
indexSearch.searchTSIDs returned localTSIDs=10
```

**Screenshots**

N/A

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./bin/victoria-metrics-prod --version
victoria-metrics-20220118-154902-tags-v1.71.0-0-g4701c108-dirty-83a76bbb
```

**Used command-line flags**
Please provide applied command-line flags used for running VictoriaMetrics and its components. 
```
./bin/victoria-metrics-prod \
 --retentionPeriod=120 \
 --storageDataPath=./storage \
 --envflag.enable=true \
 --envflag.prefix=VM_ \
 --loggerFormat=json \
 --search.maxQueryLen=65536 \
 --search.noStaleMarkers=true
```

**Comment by immerrr** [comment]
I have just found https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1899, and one of the suggestions was to remove 
 `metadata/minTimestampForCompositeIndex` file. Since the bug was about composite index/tags, and we did upgrade from v1.54 at some point(I don't remember going back to lower version, but it is possible), I tried the suggested fix to remove the min timestamp file, and it worked like a charm locally.

Then I proceeded to do the same for our staging & prod instances, and looks like it worked.
